test/unittest/component/test_listener.py

```diff
@@ -172,3 +189,54 @@ def test_listener_cleanup(db, data):
 
     db.remove('listener', listener1.identifier, force=True)
     assert not db.databackend.check_output_dest(listener1.predict_id)
+
+
+def test_listener_chaining_with_trainer(db):
+    db.cfg.auto_schema = True
+    table = db['test']
+
+    def insert_random(start=0):
+        data = []
+        for i in range(5):
+            y = int(random.random() > 0.5)
+            data.append(
+                Document(
+                    {
+                        "x": i + start,
+                        "y": y,
+                    }
+                )
+            )
+
+        db.execute(table.insert(data))
+
+    # Insert data
+    insert_random()
+
+    features = ObjectModel("features", object=lambda x: x + 1)
+
+    trainable_model = _Tmp(identifier="trainable_model", object=lambda x: x + 2)
+
+    features_listener = Listener(
+        model=features,
+        select=table.select(),
+        key="x",
+        identifier="listener1",
+        uuid="listener1",
+    )
+    features_listener.db = db
```

Comment:
Shouldn't be necessary with `db.apply`

Reply:
`features_listener.outputs_select`
will fail